references/embedding/sampler.py

```diff
@@ -0,0 +1,73 @@
+import torch
+from torch.utils.data.sampler import Sampler
+from collections import defaultdict
+import random
+
+
+def create_groups(groups, k):
+    """Bins sample indices with respect to groups, remove bins with less than k samples
+
+    Args:
+        groups (list[int]): where ith index stores ith sample's group id
+
+    Returns:
+        defaultdict[list]: Bins of sample indices, binned by group_idx
+    """
+    group_samples = defaultdict(list)
+    for sample_idx, group_idx in enumerate(groups):
+        group_samples[group_idx].append(sample_idx)
+
+    keys_to_remove = []
+    for key in group_samples:
+        if len(group_samples[key]) < k:
+            keys_to_remove.append(key)
+            continue
+
+    for key in keys_to_remove:
+        group_samples.pop(key)
+
+    return group_samples
+
+
+class PKSampler(Sampler):
+    """
+    Randomly samples from a dataset  while ensuring that each batch (of size p * k)
+    includes samples from exactly p labels, with k samples for each label.
+
+    Args:
+        groups (list[int]): List where the ith entry is the group_id/label of the ith sample in the dataset.
+        p (int): Number of labels/groups to be sampled from in a batch
+        k (int): Number of samples for each label/group in a batch
+    """
+
+    def __init__(self, groups, p, k):
+        self.p = p
+        self.k = k
+        self.groups = create_groups(groups, self.k)
+
+    def __iter__(self):
```

Comment:
This is ok as is because we are not yet adding samplers to the main library, but once we move it to the `torchvision` package, it would be good to have tests for it.

If you could write a basic test now checking the behavior (with dummy data), it would make it much easier for moving this to torchvision core later on.

Reply:
I have written the test cases for the sampler and refactored the accuracy section into another method.

I don't think we should assume that a targets attribute exists. Instead I feel like I can explain the targets data structure better, so that users can construct targets accordingly (or use the targets attribute if it exists). I have changed the comments slightly, explaining what is expected of the targets variable. Will this be okay instead?